File: src/types/backup.ts
export type BackupType = "full" | "database";
export type BackupStatus = "success" | "failed";

export interface Backup {
  id: number;
  file_name: string;
  backup_type: BackupType;
  file_path: string;
  size: number;
  status: BackupStatus;
  created_by: number | null;
  created_at: string;
}

export function parseBackupTime(input: string): Date {
  return new Date(input.replace(" ", "T") + "Z");
}

export function formatBytes(bytes: number): string {
  if (!bytes || bytes <= 0) return "0 B";
  const units = ["B", "KB", "MB", "GB"];
  const i = Math.min(units.length - 1, Math.floor(Math.log(bytes) / Math.log(1024)));
  const value = bytes / Math.pow(1024, i);
  return `${value.toFixed(value >= 100 || i === 0 ? 0 : 1)} ${units[i]}`;
}

File: src-tauri/src/models/backup.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "lowercase")]
pub enum BackupType {
    Full,
    Database,
}

impl BackupType {
    pub fn as_str(&self) -> &'static str {
        match self {
            BackupType::Full => "full",
            BackupType::Database => "database",
        }
    }

    pub fn from_str(s: &str) -> Self {
        match s {
            "full" => BackupType::Full,
            _ => BackupType::Database,
        }
    }
}

#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "lowercase")]
pub enum BackupStatus {
    Success,
    Failed,
}

impl BackupStatus {
    pub fn as_str(&self) -> &'static str {
        match self {
            BackupStatus::Success => "success",
            BackupStatus::Failed => "failed",
        }
    }

    pub fn from_str(s: &str) -> Self {
        match s {
            "failed" => BackupStatus::Failed,
            _ => BackupStatus::Success,
        }
    }
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Backup {
    pub id: i64,
    pub file_name: String,
    pub backup_type: BackupType,
    pub file_path: String,
    pub size: i64,
    pub status: BackupStatus,
    pub created_by: Option<i64>,
    pub created_at: String,
}